File: arrays/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const numsLength = nums.length;
  const diffMap = {};
  for (let i = 0; i < numsLength; i++) {
    const diff = target - nums[i];
    if (diffMap[nums[i]] > -1) {
      return [diffMap[nums[i]], i];
    } else {
      diffMap[diff] = i;
    }
  }
};

//console.log(twoSum([2,7,11,15], 9));

File: arrays/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> diffMap = new HashMap<>();
        for(int i=0;i<nums.length;i++) {
            int diff=target-nums[i];
            if(diffMap.containsKey(diff))
                return new int[] {diffMap.get(diff),i};
            diffMap.put(nums[i],i);   
        }
        return new int[]{};
    }
}
